app/controllers/api/v1/publications_controller.rb

```diff
@@ -0,0 +1,314 @@
+          end
+          response 200 do
+            key :description, 'Successful update of Publication object'
+            schema do
+              key :title, 'Publication'
+              key :'$ref', :Publication
+            end
+          end
+          response 401 do
+            key :description, ApiBaseController.unauthorized
+          end
+          response 403 do
+            key :description, ApiBaseController.forbidden('edit Publication')
+          end
+          response 404 do
+            key :description, ApiBaseController.not_found(Study, Publication)
+          end
+          response 406 do
+            key :description, ApiBaseController.not_acceptable
+          end
+          extend SwaggerResponses::ValidationFailureResponse
+        end
+      end
+
+      # PATCH /single_cell/api/v1/studies/:accession/publications/:id
+      def update
+        sanitized_update_params = publication_params.to_unsafe_hash.keep_if {|k,v| !v.blank?}
+        if @publication.update(sanitized_update_params)
+          render :show
+        else
+          render json: {errors: @publication.errors}, status: :unprocessable_entity
+        end
+      end
+
+      swagger_path '/studies/{accession}/publications/{id}' do
+        operation :delete do
+          key :tags, [
+              'Publications'
+          ]
+          key :summary, 'Delete an Publication'
+          key :description, 'Deletes a single Publication'
+          key :operationId, 'delete_study_publication_path'
+          parameter do
+            key :name, :accession
+            key :in, :path
+            key :description, 'Accession of Study'
+            key :required, true
+            key :type, :string
+          end
+          parameter do
+            key :name, :id
+            key :in, :path
+            key :description, 'ID of Publication to delete'
+            key :required, true
+            key :type, :string
+          end
+          response 204 do
+            key :description, 'Successful Publication deletion'
+          end
+          response 401 do
+            key :description, ApiBaseController.unauthorized
+          end
+          response 403 do
+            key :description, ApiBaseController.forbidden('delete Publication')
+          end
+          response 404 do
+            key :description, ApiBaseController.not_found(Study, Publication)
+          end
+          response 406 do
+            key :description, ApiBaseController.not_acceptable
+          end
+        end
+      end
+
+      # DELETE /single_cell/api/v1/studies/:accession/publications/:id
+      def destroy
+        begin
+          @publication.destroy
+          head 204
+        rescue => e
+          ErrorTracker.report_exception(e, current_api_user, @publication, params)
+          MetricsService.report_error(e, request, current_api_user, @study)
+          render json: {error: e.message}, status: 500
+        end
+      end
+
+      private
+
+      def set_study
```

Comment:
We have a concern called `StudyAware` that is included in all `ApiBaseController` descendants.  This provides built-in methods for `:set_study`, `:check_study_view_permission`, and `check_study_edit_permission`.  As this concern is already included in this controller, you can remove `:set_study` and `: check_study_permission` here and just rely on the `before_action` filters you've set up (though you'll need to rename `:check_study_permission` to `:check_study_edit_permission`.

Again, this seems to be a side effect of copying from `ExternalResoucesController`.  You can make the same edits there as well.  That controller predated the `StudyAware` module.

Reply:
Helpful!  Done in 254af54.  I also widened accessibility for public studies in 02b4b1b.